docs/Troubleshooting.md

```diff
@@ -0,0 +1,113 @@
+## Troubleshooting guide.
+Refer to this section for common issues faced while deploying your Pytorch models using Torchserve and their corresponding troubleshooting steps.
+
+* [Deployment and config issues](#deployment-and-config-issues)
+* [Snapshot related issues](#snapshot-related-issues)
+* [API related issues](#api-relate-issues)
+* [Model-archiver](#model-archiver)
+
+
+### Deployment and config issues
+#### "Failed to bind to address: http://127.0.0.1:8080", port 8080/8081 already in use.
+Usually, the port number 8080/8081 is already used by some other application or service, it can be verified by using cmd `ss -ntl | grep 8080`.  There are two ways to troubleshoot this issue
+either kill the process which is using port 8080/8081 or run Torchserve on different ports other than 8080 & 8081.
+
+Refer  [configuration.md](https://github.com/pytorch/serve/blob/master/docs/configuration.md) for more details.
+
+Relevant issues:  [[542](https://github.com/pytorch/serve/issues/542)]
+
+#### "java.lang.NoSuchMethodError" when starting Torchserve.[[473](https://github.com/pytorch/serve/issues/473)]
+This error usually occurs when Java 11 is not installed or used. Java 11 is required by Torchserve and java versions before it is not supported.
+
+Relevant issues: [[#473](https://github.com/pytorch/serve/issues/473)]
+
+####  Unable to send big files for inference request?
+The default max request size and response size is roughly 6.5 Mb. Hence any file size greater than 6.5mb cannot be uploaded.
+To resolve this update `max_request_size` and `max_response_size` in a config.properties file and start the torchserve with this config file.
+```
+$ cat config.properties
+max_request_size=<request size in bytes>
+max_response_size=<response size in bytes>
+$ torchserve --start --model-store model_store --ts-config /path/to/config.properties
+```
+You can also use environment variables to set these values.
+Refer  [configuration.md](https://github.com/pytorch/serve/blob/master/docs/configuration.md) for more details.
+Relevant issues: [[#335](https://github.com/pytorch/serve/issues/335)]
+
+###  Snapshot related issues
+#### How to disable Snapshot feature?
+By default, the snapshot feature is enabled. To disable snapshot feature start torchserve using --ncs flag or  specify config file using --ts-config path/to/config 
+Relevant issues:[[#383](https://github.com/pytorch/serve/issues/383), [#512](https://github.com/pytorch/serve/issues/512)]
+
+#### Torchserve stopped after restart with "InvalidSnapshotException" exception.
+Torchserve when restarted uses the last snapshot config file to restore its state of models and their number of workers.
+
```

Comment:
Unnecessary line breaks here and in the following sections

Reply:
Done.